s3/pkg/datastore/aws/aws.go

```diff
@@ -157,9 +214,63 @@ func (ad *AwsAdapter) Get(ctx context.Context, object *pb.Object, start int64, e
 	return result.Body, nil
 }
 
+
+func (ad *AwsAdapter) BucketDelete(ctx context.Context, in *pb.Bucket) error {
+	log.Info("Bucket delete is called in aws service")
+	Credentials := credentials.NewStaticCredentials(ad.backend.Access, ad.backend.Security, "")
+	configuration := &aws.Config{
+		Region:      aws.String(ad.backend.Region),
+		Endpoint:    aws.String(ad.backend.Endpoint),
+		Credentials: Credentials,
+	}
+
+	svc := awss3.New(session.New(configuration))
+	input := &awss3.DeleteBucketInput{
+		Bucket: aws.String(in.Name),
+	}
+
+	result, err := svc.DeleteBucketWithContext(ctx, input)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			default:
+				log.Error(aerr.Error())
+			}
+		} else {
+			// Print the error, cast err to awserr.Error to get the Code and
+			// Message from an error.
+			log.Error(err.Error())
+		}
+		return err
+	}
+
+	// Wait until bucket is gone before finishing
+	err = svc.WaitUntilBucketNotExists(&awss3.HeadBucketInput{
+		Bucket: aws.String(in.Name),
+	})
+	if err != nil {
+		return err
```

Comment:
What is the return message from the caller of this function? suppose it times out, then what is the api response?